src/scrapers/fass_scraper/crawler.py

```diff
@@ -1,225 +1,170 @@
-    # Get reference to current letter page
-    html_ref = f"https://www.fass.se/LIF/pharmaceuticallist?page={letter}"
-    soup = fetch_url(html_ref)
-
-    # Get all tags containing names of drugs
-    names = soup.select('.productResultPanel .expandcontent .innerlabel')
-    # Get all tags containing links to drug info
-    links = soup.select('.productResultPanel .expandcontent .linkList a')
-
-    # Extract drug names
-    product_names = [name.get_text().strip() for name in names]
-    # Extract drug ids
-    product_ids = [link.get('href').strip()[-14:] for link in links]    # The product link is relative 
-                                                                        # To get the absolute link we extract only the nplID of the link
-
-    return zip(product_ids, product_names)
-
-# Retrives the summary file 
-def get_summary_file():
-    summary = {}
-    if os.path.isfile(SUMMARY_FILE):
-        with open(SUMMARY_FILE, 'r') as infile:
-            summary = json.load(infile)
-    return summary
-
-
-# Crawls through all the medecines in the alpabetical list of medecines found on
-# https://www.fass.se/LIF/pharmaceuticalliststart?userType=2
-def crawl_alphabetical_list():
-    ALPHABET = "ABCDEFGHIJKLMNOPQRSTUVWXYZÅÄÖ"
-    PRODUCT_BASE_LINK = "https://www.fass.se/LIF/product?nplId="
+        return BeautifulSoup(response.text, 'lxml', parse_only=strainer)
+
+    return None
+
 
 
-    # Load existing summary if it exists
-    summary = get_summary_file()
+def assert_content(result):
+    original = {}
+    with open(f"../data/products/{result[0]}.json", "r") as doc:
+        original = json.load(doc)
+    if original != result[1]:
+        print(f"{result[0]} Failed")
+        with open(f"{result[0]}.json", "w") as doc:
+            json.dump(result[1], doc, indent=4, ensure_ascii=False)
 
-    # Create the 'products' directory if it doesn't exist
-    if not os.path.exists(PRODUCT_DIR):
-        os.makedirs(PRODUCT_DIR)
+    else:
+        print(f"{result[0]} Successful")
 
 
-    for page in PAGES.keys():
-        keys[page] = []
+class MedicalPage:
+    nplid = None
+    name = None
+    links = []
 
-    count = 0
-    for letter in ALPHABET:
-        # Iterate over every product beginning with current letter
-        for (product_id, product_name) in get_listed_products_from_alphabetical(letter):
-            count += 1
-            full_url = PRODUCT_BASE_LINK + product_id
-            print(f"*****************************\n{full_url} {count}\n********************************")
+    responses = []
+    json = {}
 
-            # Check if this product has been fully downloaded
-            if summary.get(product_id, {}).get('fully_downloaded', False):
+    PRODUCTS_DIR = "../data/products/"
+
+    def __init__(self, nplid, name, baseLink):
+        self.nplid = nplid
+        self.name = name
+
+        docTypes = [7, 6, 3, 2000, 78, 80]
+        self.links = [baseLink + f"&docType={page}" for page in docTypes]
```

Comment:
missing: labels for page numbers for later debugging 
```suggestion
        # Bipacksedel: 7, Produktresume: 6, Förpackningar: 30, Fass_text: 3, Bilder_och_delbarhet: 2000, Miljöinformation: 78, Skyddsinfo: 80
        docTypes = [7, 6, 3, 2000, 78, 80] 
```

Reply:
Accepted via ffc01d17c85c75fc8ce5c11f768b437274163755